internal/data/spicedb.go

```diff
@@ -242,6 +244,50 @@ func (s *SpiceDbRepository) LookupResources(ctx context.Context, resouce_type *a
 	return resources, errs, nil
 }
 
+func (s *SpiceDbRepository) ImportBulkTuples(stream grpc.ClientStreamingServer[apiV1beta1.ImportBulkTuplesRequest, apiV1beta1.ImportBulkTuplesResponse]) error {
+	if err := s.initialize(); err != nil {
+		return err
+	}
+
+	var totalImported uint64
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			if !errors.Is(err, io.EOF) {
+				if err := stream.SendAndClose(&apiV1beta1.ImportBulkTuplesResponse{NumImported: totalImported}); err != nil {
+					return status.Errorf(codes.Internal, "failed to send response: %v", err)
+				}
+			}
+			return err
+		}
+		inputRelationships := (*req).Tuples
+		batch := []*v1.Relationship{}
+		for _, tuple := range inputRelationships {
+			tuple.Relation = addRelationPrefix(tuple.Relation, relationPrefix)
+			batch = append(batch, createSpiceDbRelationship(tuple))
+		}
+		client, err := s.client.ImportBulkRelationships(context.Background())
```

Comment:
If I am reading this right, it looks like this method...

1. Gets a batch from the input stream
2. Starts a streaming RPC with SpiceDB
3. Sends one batch on that stream
4. Closes that stream

And then repeats for each batch.

However what we want is:

1. Start the spicedb stream
2. For each batch, send that batch to SpiceDB
3. Then close the stream after there are no more input batches (the client closes the stream on Relations end)

Reply:
Good catch, thank you for the suggest. I update the code.